presto-main/src/test/java/com/facebook/presto/operator/BenchmarkWindowOperator.java

```diff
@@ -0,0 +1,330 @@
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jmh.runner.options.VerboseMode;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ScheduledExecutorService;
+
+import static com.facebook.presto.SessionTestUtils.TEST_SESSION;
+import static com.facebook.presto.block.BlockAssertions.createLongSequenceBlock;
+import static com.facebook.presto.operator.BenchmarkWindowOperator.Context.ROWS_PER_PAGE;
+import static com.facebook.presto.operator.BenchmarkWindowOperator.Context.TOTAL_PAGES;
+import static com.facebook.presto.operator.TestWindowOperator.ROW_NUMBER;
+import static com.facebook.presto.operator.TestWindowOperator.createFactoryUnbounded;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static io.airlift.concurrent.Threads.daemonThreadsNamed;
+import static io.airlift.units.DataSize.Unit.GIGABYTE;
+import static java.util.concurrent.Executors.newCachedThreadPool;
+import static java.util.concurrent.Executors.newScheduledThreadPool;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.openjdk.jmh.annotations.Mode.AverageTime;
+import static org.openjdk.jmh.annotations.Scope.Thread;
+import static org.testng.Assert.assertEquals;
+
+@State(Thread)
+@OutputTimeUnit(MILLISECONDS)
+@BenchmarkMode(AverageTime)
+@Fork(3)
+@Warmup(iterations = 5)
+@Measurement(iterations = 10, time = 2, timeUnit = SECONDS)
+public class BenchmarkWindowOperator
+{
+    @State(Thread)
+    public static class Context
+    {
+        public static final int NUMBER_OF_GROUP_COLUMNS = 2;
+        public static final int TOTAL_PAGES = 140;
+        public static final int ROWS_PER_PAGE = 10000;
+        static final List<Integer> PARTITION_CHANNELS = Ints.asList(0, 1);
+
+        @Param({"10", "20", "100"})
+        public int rowsPerPartition;
+
+        @Param({"0", "1", "2", "3"})
+        public int numberOfPregroupedColumns;
+
+        @Param({"10", "50", "100"})
+        public int partitionsPerGroup = 10;
+
+        private ExecutorService executor;
+        private ScheduledExecutorService scheduledExecutor;
+        private OperatorFactory operatorFactory;
+
+        private List<Page> pages;
+
+        @Setup
+        public void setup()
+        {
+            executor = newCachedThreadPool(daemonThreadsNamed("test-executor-%s"));
+            scheduledExecutor = newScheduledThreadPool(2, daemonThreadsNamed("test-scheduledExecutor-%s"));
```

Comment:
This will create an executor for each benchmark thread. What is this executor used for? It doesn’t seem necessary for an operator benchmark. We may need to change how the taskcontext object is created to not need it.